docs/source/developers/python.rst

```diff
@@ -566,6 +567,14 @@ Now, we can build pyarrow:
    For building pyarrow, the above defined environment variables need to also
    be set. Remember this if to want to re-build ``pyarrow`` after your initial build.
 
+.. note::
+
+   In case you are following this steps and are therefore using Python
+   version < 3.10 together with Conda, then ``CONDA_DLL_SEARCH_MODIFICATION_ENABLE``
```

Comment:
We should maybe update the example to use Python 3.10? 

We can still keep this note, but then at least it is not the default thing people run into when copy-pasting our examples

Reply:
Perhaps we can word this note so it doesn't need to be updated later?

```
   If you are using Conda with Python 3.9 or earlier, you must
   set ``CONDA_DLL_SEARCH_MODIFICATION_ENABLE=1``.
```

Reply:
That's fine by me +1